addons/mail/static/src/core/common/message.js

```diff
@@ -511,4 +517,32 @@ export class Message extends Component {
             message: this.props.message,
         });
     }
+
+    // TODO KFR: Create something like a translation data object
+    async attemptTranslation() {
+        const response = await this.rpc("/discuss/translation/translate", {body: this.message.body})
+        if (response?.value) {
+            this.message.translatedBody = markup(response.value);
+            this.message.translationSource = response.source;
+        }
+        else {
+            this.message.translationErrors = true;
+        }
+    }
+
+    async onClickTranslate() {
+        if (!this.message.translatedBody) {
+            await this.attemptTranslation();
+        }
```

Comment:
The function split `attemptTranslation` is not worth it when called only once, and the caller is trivial.

Reply:
The function has been removed